app/presenters/user_applications_presenter.rb

```diff
@@ -4,10 +4,13 @@ def initialize(user)
   end
 
   def user_applications
-    @applications ||= @user.o_auth_applications.map do |app|
-      authorization = Api::OpenidConnect::Authorization.find_by_client_id_and_user(app.client_id, @user)
-      UserApplicationPresenter.new app, authorization.scopes, authorization.id
+    application_array = @user.o_auth_applications.map do |app|
+      authorizations = Api::OpenidConnect::Authorization.find_all_by_client_id_and_user(app.client_id, @user)
+      authorizations.map {|authorization|
+        UserApplicationPresenter.new app, authorization.scopes, authorization.id
+      }
     end
+    @applications ||= Array(application_array.flatten!)
   end
```

Comment:
This should be possible with a single query actually, something like `Api::OpenidConnect::Authorization.joins(:o_auth_application).where(user: @user, o_auth_application: {user: @user}).map` perhaps?